src/gui/options_imp.cpp

```diff
@@ -1079,6 +1093,26 @@ void options_imp::enableApplyButton()
     applyButton->setEnabled(true);
 }
 
+void options_imp::adaptDownloadLimitStep(QString s)
```

Comment:
>can you use valueChanged(int) here?

If not, then change `QString s` to `const QString &s`
and for below similar functions.